v2/codetags/parser.go

```diff
@@ -183,139 +206,142 @@ func parseTagKey(input string) (TypedTag, error) {
 		cur.Name = buf.String()
 		buf.Reset()
 	}
+	saveTag := func() error {
+		usingNamedArgs := false
+		for i, arg := range args {
+			if (usingNamedArgs && arg.Name == "") || (!usingNamedArgs && arg.Name != "" && i > 0) {
+				return fmt.Errorf("can't mix named and positional arguments")
+			}
+			if arg.Name != "" {
+				usingNamedArgs = true
+			}
+		}
+		if !usingNamedArgs && len(args) > 1 {
+			return fmt.Errorf("multiple arguments must use 'name: value' syntax")
+		}
 
-	runes := []rune(input)
+		newTag := &TypedTag{Name: tag.String(), Args: args}
+		if startTag == nil {
+			startTag = newTag
+			endTag = newTag
+		} else {
+			endTag.ValueTag = newTag
+			endTag.ValueType = ValueTypeTag
+			endTag = newTag
+		}
+		args = nil // Reset to nil instead of empty slice
+		tag.Reset()
+		return nil
+	}
+	saveValue := func() {
+		endTag.Value = value.String()
+		if opts.rawValues {
+			endTag.ValueType = ValueTypeRaw
+			return
+		}
+		endTag.ValueType = valueType
+		if valueType == ValueTypeString && (endTag.Value == "true" || endTag.Value == "false") {
+			endTag.ValueType = ValueTypeBool
+		}
+	}
 	st := stBegin
 parseLoop:
-	for i, r = range runes {
+	for r := s.peek(); r != EOF; r = s.peek() {
 		switch st {
 		case stBegin:
 			switch {
 			case unicode.IsSpace(r):
+				s.next()
 				continue
 			case isIdentBegin(r):
-				tag.WriteRune(r)
 				st = stTag
 			default:
 				break parseLoop
 			}
 		case stTag:
 			switch {
-			case isIdentInterior(r) || r == ':':
-				tag.WriteRune(r)
+			case isIdentBegin(r):
+				ident, err := s.nextIdent(isTagNameInterior)
+				if err != nil {
+					return TypedTag{}, err
+				}
+				tag.WriteString(ident)
+				st = stMaybeArgs
+			default:
+				break parseLoop
+			}
+		case stMaybeArgs:
+			switch {
 			case r == '(':
+				s.next()
 				incomplete = true
 				st = stArg
 			case r == '=':
+				s.next()
+				hasValue = true
```

Comment:
should this set incomplete?  If you hit EOF right after `=` it should be an error, right?

Reply:
I'll add the check for typed values.  For ParseRaw, I'll allow an empty string value.